shoop_tests/core/test_fields.py

```diff
@@ -7,10 +7,38 @@
 # LICENSE file in the root directory of this source tree.
 import pytest
 from django.core.exceptions import ImproperlyConfigured
+from django.forms import ModelForm
 
 from shoop.core.fields import MeasurementField
+from shoop.core.models import Product
+from shoop.testing.factories import get_default_product, get_default_shop
 
 
 def test_measurement_field_doesnt_know_bananas():
     with pytest.raises(ImproperlyConfigured):
         scale = MeasurementField(unit="banana")
+
+@pytest.mark.django_db
+def test_formatted_decimal_field():
+    """
+    Test that FormattedDecimalField doesn't return value in scientific
+    notation.
+    """
+
+    class TestModelForm(ModelForm):
+        class Meta:
+            model = Product
+            fields = ["width"]
+
+    get_default_shop()
+
+    product = get_default_product()
+    product.width = 0
+    product.save()
+    product = Product.objects.first()
+
+    form = TestModelForm(instance=product)
+    rendered_form = str(form)
+
+    assert "value=\"0E-9\"" not in rendered_form
+    assert "value=\"0\"" in rendered_form
```

Comment:
Hmm.. having second thought about the truncation of long values, since that could lead to errors which are hard to find. Maybe we should just raise if the string representation would be too large. This could do it:

```python
def format_decimal(value, max_digits=100, exponent_limit=100)
    assert isinstance(value, decimal.Decimal)
    val = value.normalize()
    (sign, digits, exponent) = val.as_tuple()
    if exponent > exponent_limit:
        raise ValueError('Exponent too large for formatting: %r' % value)
    elif exponent < -exponent_limit:
        raise ValueError('Exponent too small for formatting: %r' % value)
    if len(digits) > max_digits:
        raise ValueError('Too many digits for formatting: %r' % value)
    return format(val, 'f')
```